doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,480 @@
+   Google Highway with our own numerical routines on top of that. It does not
+   (yet) assume a decision or proposed decision is made. Hence this NEP is not
+   "this is proposed" with another option in the Alternatives section, but
+   rather a side-by-side comparison.
+
+    
+Development effort and long-term maintainability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Moving to Highway is likely to be a significant development effort.
+Longer-term, this will hopefully be offset by Highway itself having more
+maintainer bandwidth to deal with ongoing issues in compiler support and adding
+new platforms. It should increase the bus factor of SIMD support in NumPy to
+some extent. That said - the Highway repository also seems to show that its bus
+factor is low - it was 1 until the start of 2023, and now it's perhaps 2.
+
+Highway being used by other projects, like Chromium and `JPEG XL`_ (see
+`this more complete list <https://google.github.io/highway/en/master/README.html#examples>`__
+in the Highway documentation), does imply that there is likely to be a benefit
+of a wider range of testing and bug reporting/fixing.
+
+One concern is that new instructions may have to be added, and that that is
+often best done as part of the process of developing the numerical kernel that
+needs the instruction. This will be a little more clumsy if the instruction
+lives in Highway which is a git submodule inside the NumPy repo - there will be
+a need to implement a temporary/generic version first, and then update the
+submodule after upstreaming the new intrinsic.
+https://numpy.org/doc/1.25/reference/simd/
+
+Documentation-wise, Highway would be a clear win. NumPy's
+`CPU/SIMD Optimizations`_ docs are fairly sparse compared to
+`the Highway docs`_.
+
+
+Licensing
+~~~~~~~~~
+
+Highway is licensed under Apache 2.0, which is incompatible with NumPy's BSD-3
+license. Apache 2.0 is not a copyleft license, hence if we'd use Highway as a
+build-time dependency only, that would have no impact on NumPy's licensing situation.
+However, we don't want to add a new build-time dependency - at least certainly
+not before we have removed all traces of the ``setup.py``-based build (it's a lot
+easier with Meson). Pulling in Highway as a git submodule and shipping it in sdist's
+would change NumPy's license status.
+
+That license status change would still be manageable perhaps, if it'd be contained to
+one git submodule that redistributors can swap out for an external dependency though.
+However, there is a second issue: reusing code snippets and potentially larger chunks
+of code from Highway itself, its docs, or other Highway-using projects would not work
+without changing NumPy's licensing status to a mix of BSD-3/MIT and Apache 2.0.
+
+As of now (July 2023), the only Apache 2.0 usages inside the NumPy repository are
+the vendored ``mathjax`` code to build the docs, and the code for the PCG64
+random number generator which was relicensed as MIT for NumPy's use.
```

Comment:
Now that Highway is dual-licensed I will remove the above section.

Reply:
I think putting in at least a footnote that Highway went through a relicensing to make it easier for MIT/BSD-licensed projects like NumPy to adopt it, in response to this NEP, would make sense. It's part of the history of the discussions around this NEP, and I'm sure it took a bit of talking to get it done.

Reply:
OK, I will add.